nixpkgs/fetchNixpkgs.nix

```diff
@@ -0,0 +1,51 @@
+{ rev                             # The Git revision of nixpkgs to fetch
+, sha256                          # The SHA256 of the downloaded data
+, system ? builtins.currentSystem # This is overridable if necessary
+}:
+
+with {
+  ifThenElse = { bool, thenValue, elseValue }: (
+    if bool then thenValue else elseValue);
+};
```

Comment:
Just curious: why define this rather than using the built-in if/then/else syntax?

Reply:
If you want to name the large branches you could do `let trueBranch = ....; falseBranch = ...; in if bool then trueBranch else falseBranch`.

Reply:
I got this off the NixOS wiki page, I believe @taktoa wrote it.

Reply:
In this case, at least in my opinion, I believe it increases the readability of the nix file.

Reply:
My 2c would be to use something standard like `let` to improve readability rather than coming up with a custom function. _@3noch_ goes and modifies the NixOS wiki real quick..._ 😛